sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -935,6 +935,105 @@ def sdc_pandas_dataframe_drop_impl(df, labels=None, axis=0, index=None, columns=
     return func_def, global_vars
 
 
+def _dataframe_reduce_columns_codegen_isna(func_name, func_params, columns, df):
+    results = []
+    joined = ', '.join(func_params)
+    func_lines = [f'def _df_{func_name}_impl({joined}):']
+    ind = df_index_codegen_all(df)
+    for i, c in enumerate(columns):
+        result_c = f'result_{c}'
+        func_lines += [f'  series_{c} = pandas.Series(get_dataframe_data({func_params[0]}, {i}))',
+                       f'  {result_c} = series_{c}.{func_name}()']
+        results.append((columns[i], result_c))
+
+    data = ', '.join(f'"{col}": {data}' for col, data in results)
+    func_lines += [f'  return pandas.DataFrame({{{data}}}{ind})']
```

Comment:
The index value is not clear. May be it is better to define index code line here?
```suggestion
    index = 'df.index'
    func_lines.append(f"  return pandas.DataFrame({{{data}}}, index={index})")
```